Source/WebKit/UIProcess/Notifications/glib/NotificationService.cpp

```diff
@@ -61,6 +67,48 @@ namespace WebKit {
 
 static const Seconds s_dbusCallTimeout = 20_ms;
 
+#if USE(SKIA)
+
+// Alias to avoid conflicting with write() below.
+ssize_t (*writeToFD)(int, const void*, size_t) = write;
+
+// Simple stream that writes to an fd.
+class fdWStream : public SkWStream {
```

Comment:
Wouldn't it be easier to just close the fd and use `SkFILEWStream` with the given path?

Reply:
No, because then it becomes racey. We have a valid FD.

Reply:
> Wouldn't it be easier to just close the fd and use SkFILEWStream with the given path?

Surely that's worse than what Patrick has done here.

Reply:
Can g_file_open_tmp return the same file path? I didn't say it's better, just that it's a lot easier. `FileSystem::createTemporaryFile()` does that, just closing the fd.

Reply:
No, I'm ok if you both insist it's better this way.